community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/util.py

```diff
@@ -181,22 +181,24 @@ def sockets(self) -> int:
 
 @dataclass(frozen=True)
 class UpcomingMaintenance:
-    window_start_time: datetime
+    type: str
+    window_start_time: Optional[datetime]
 
     @classmethod
     def from_json(cls, jo: Optional[dict]) -> Optional["UpcomingMaintenance"]:
         if jo is None:
             return None
         try:
+            type = jo["type"]
             if "windowStartTime" in jo:
                 ts = parse_gcp_timestamp(jo["windowStartTime"])
             elif "startTimeWindow" in jo:
                 ts = parse_gcp_timestamp(jo["startTimeWindow"]["earliest"])
             else:
-                raise Exception("Neither windowStartTime nor startTimeWindow are found")
+                ts = None
         except BaseException as e:
             raise ValueError(f"Unexpected format for upcomingMaintenance: {jo}") from e
```

Comment:
A couple broad concerns:

- when I have seen maintenance parsing errors in logs, it ends up printing a stacktrace and it doesn't help guide to the specific problem
- the type of errors here seems relatively simple (KeyError for missing `type`, invalid format for timestamp).
- does raising this exception block subsequent execution of slurmsync? The fact that stacktrace is printed suggests to my eyes that it does

Reply:
> does raising this exception block subsequent execution of slurmsync?

No, see

```py
log.exception("Failed to parse upcomingMaintenance, ignoring")
maint = None # intentionally swallow exception
```